support/rally.js

```diff
@@ -93,6 +116,22 @@ export default class Rally {
     }
   }
 
+  /**
+   * Pause the current study.
+   */
+  _pause() {
+    this._stateChangeCallback("pause");
+    this._state = STATES.PAUSED;
```

Comment:
My feeling is that checking to see if the study is paused is additional message overhead (the core only knows that the study *should* be paused, we need to actually send a message to the study to determine its state.)

Sending a single message to the study informing it that it should pause makes sense in this context, since only the study has this state. I might be wrong though I'll think about this a bit more :)

Reply:
I mean just

```suggestion
  _pause() {
    if (this._state !== STATES.PAUSED) {
    this._stateChangeCallback("pause");
    this._state = STATES.PAUSED;
    }
```

just to make sure studies don't pause twice if we make mistakes in the core addon and send the message twice. The rationale is that we control the quality of the code in the core add-on, but we can't guarantee that a double pause won't create problems in the study right now (

Reply:
Oh shoot right for some reason I thought you were pointing at the `Core.js` and not `rally.js` :) of course this sounds good ty.